logic/marathonScheduler26.js

```diff
@@ -0,0 +1,504 @@
+      " = " +
+      new Date(runnerData.raceDate)
+  );
+
+  //calculate time diff between training start date and last training date
+  const timeDifference = lastRegularTrainingDay - runnerData.startDate;
+  // calculate the no. of days between two dates by dividing by a day of milliseconds
+  const dayDifference = timeDifference / (1000 * 3600 * 24);
+  logthis("total days of training time = " + dayDifference);
+  // calculate totalweeks to train
+  const weeksToTrain = Math.ceil(dayDifference / 7);
+  logthis("total weeks of training time = " + weeksToTrain);
+
+  const weeksToIncrementMiles = Math.ceil((dayDifference / 7) * 0.666);
+  logthis(
+    "total weeks of incrementations (2/3 weeks) = " + weeksToIncrementMiles
+  );
+
+  // calculate totalTrainingSessions
+  const totalSessions = runDays.length * weeksToTrain;
+  logthis("total run sessions (weeks x perweek) = " + totalSessions);
+
+  // figure out if we are going to exceed max miles per week with default weeklyIncrement miles per week increase..********************************************
+  // If we exceed peak, then adjust the weeklyIncrement downward to meet peak goal.
+  let weeklyIncrement = 3;
+  // maxMilesPerWeek limited to 60 if only training 3 days per week.
+  let maxMilesPerWeek;
+  if (runDays.length < 4) {
+    // maxMilesPerWeek limited to 60 if only training 3 days per week.
+    maxMilesPerWeek = 60;
+  } else {
+    maxMilesPerWeek = 65;
+  }
+
+  logthis(
+    "maxMilesPerWeek Goal is = " +
+      maxMilesPerWeek +
+      ",THIS schedule will reach: " +
+      maxMilesPerWeek * weeksToIncrementMiles +
+      runnerData.startMilesPerWeek
+  );
+
+  if (
+    weeksToIncrementMiles * weeklyIncrement + runnerData.startMilesPerWeek >
+    maxMilesPerWeek
+  ) {
+    // Compute new weeklyIncrement to reach max miles per week in the number of weeks of training.
+    weeklyIncrement =
+      (maxMilesPerWeek - runnerData.startMilesPerWeek) / weeksToIncrementMiles;
+  }
+
+  logthis(
+    "weeklyIncrement adjusted down if runner exceeds maxMilesPerWeek = " +
+      weeklyIncrement
+  );
+
+  const peakMiles =
+    weeklyIncrement * weeksToIncrementMiles + runnerData.startMilesPerWeek;
+  logthis("The peak miles per week will reach " + peakMiles);
+
+  // Calculate and assign running schedule and percent for each day of run
+  if (runDays.length > 6 || runDays.length < 3) {
+    logthis(
+      "ERROR--- Only " +
+        runDays.length +
+        " Run Days Chosen.. This Must be between 3 and 6. Continuing with assumed 4 days: Mon, Tue, Wed, Thu, Sat"
+    );
+    runDays.push(1);
+    runDays.push(2);
+    runDays.push(4);
+    runDays.push(6);
+  }
+
+  if (runDays.length === 3) {
+    day[0] = {
+      percentMilesPerWeek: 25,
+      description:
+        "Speed Development.  Developing speed will set you up for success on race day.  They serve as both physical and mental conditioning. Speed workouts should be executed at harder effort than race pace."
```

Comment:
Nitpick, can you save the repeating descriptions as variables and then reference the variable instead of the string each time?